spec/subjects/replay-subject-spec.js

```diff
@@ -206,4 +206,21 @@ describe('ReplaySubject', function () {
       expectObservable(subscriber1).toBe(expected1);
     });
   });
+
+  it('should not leak memory if events are fed synchronously indefinitely', function () {
+    var replaySubject = new ReplaySubject(1);
+    var event = 'foobar'.repeat(10000).split(''); // memory-hungry object
+
+    replaySubject.next(event.slice());
+    global.gc();
+    var memoryBefore = process.memoryUsage().heapUsed;
+
+    for (var i = 0; i < 100; i++) {
+      replaySubject.next(event.slice());
+    }
+    global.gc();
+    var memoryAfter = process.memoryUsage().heapUsed;
+
+    expect(memoryAfter / memoryBefore).toBeLessThan(1.2);
```

Comment:
I'm a little concerned about the potential for brittleness here in this test. Our tests could start failing because of a defect in a newly deployed runtime.